### PR TITLE
feat(landing folder): event view mode created for landing | EEF-63

### PR DESCRIPTION
**Pull request by Stanislav-Vyhnan**
EEF-63

dev
## GitHub Project

* [Main GitHub Project ticket](https://github.com/ita-social-projects/EventsExpress-Frontend/projects/1)

## Code reviewers

- [ ] All

### Second Level Review

- [ ] @VadymKharyton | @andrii-kuzmichov 

## CHECK LIST
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: src/components/helpers/landingUtils.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React from "react";
+import SliderForEvents from "../landing/SliderForEvents/SliderForEvents";
+import EventsListItem from "../landing/EventsListItem/EventsListItem";
+import EventMatrix from "../landing/EventMatrix/EventMatrix";
+
+const viewModeSwitcher = (events, viewMode) => {
+  switch (viewMode) {
+    case "slider":
+      return <SliderForEvents events={events} />;
+    case "list":
```
Comment (andrii-kuzmichov) [inline src/components/helpers/landingUtils.js]: Create constants for `slider` and `list`

File: src/components/helpers/landingUtils.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import React from "react";
+import SliderForEvents from "../landing/SliderForEvents/SliderForEvents";
+import EventsListItem from "../landing/EventsListItem/EventsListItem";
+import EventMatrix from "../landing/EventMatrix/EventMatrix";
+
+const viewModeSwitcher = (events, viewMode) => {
+  switch (viewMode) {
```
Comment (andrii-kuzmichov) [inline src/components/helpers/landingUtils.js]: Maybe refactor `switch...case` to object render

File: src/components/landing/EventsListItem/EventListCard/EventListCard.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+const VIEW_DETAIL = "View detail";
+
+const EventListCard = ({ event }) => {
+  const { id, data, title, description, location, dateFrom } = event;
+
+  const [eventImage, setEventImage] = useState(eventDefaultImage);
+
+  useEffect(() => {
+    photoService.getFullEventPhoto(data?.id).then(eventFullImage => {
+      if (eventFullImage !== null) {
```
Comment (andrii-kuzmichov) [inline src/components/landing/EventsListItem/EventListCard/EventListCard.js]: Why didn't check `undefined` case?

File: src/components/landing/EventsListItem/EventListCard/EventListCard.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        setEventImage(URL.createObjectURL(eventFullImage));
+      }
+    });
+
+    return () => URL.revokeObjectURL(eventImage);
+  }, [data?.id, eventImage]);
+
+  return (
+    <div className="card__item">
+      <div className="card__item_photo">
```
Comment (andrii-kuzmichov) [inline src/components/landing/EventsListItem/EventListCard/EventListCard.js]: extra div

File: src/components/landing/EventsListItem/EventListCard/EventListCard.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        <img src={eventImage} alt="Event_photo" />
+      </div>
+      <div className="card__item_content">
+        <h2 className="card__item_header">{title}</h2>
+        <div className="card__item_description">
```
Comment (andrii-kuzmichov) [inline src/components/landing/EventsListItem/EventListCard/EventListCard.js]: Maybe tag `<p></p>`?

File: src/components/landing/EventsViewMode/EventsViewMode.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
+import React from "react";
+import PropTypes from "prop-types";
+import "./EventsViewMode.scss";
+
+const EventsViewMode = ({ setViewMode }) => {
+  const MAPPER = [
+    { id: 1, control: `menu-view-mode__matrix`, viewMode: "matrix" },
+    { id: 2, control: `menu-view-mode__list`, viewMode: "list" },
+    { id: 3, control: `menu-view-mode__slider`, viewMode: "slider" },
+  ];
```
Comment (andrii-kuzmichov) [inline src/components/landing/EventsViewMode/EventsViewMode.js]: Move to constants

File: src/components/landing/EventsViewMode/EventsViewMode.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+        >
+          <button
+            type="button"
+            name={control}
+            id={control}
+            onClick={() => {
+              setViewMode(viewMode);
+            }}
+          >
+            {" "}
```
Comment (andrii-kuzmichov) [inline src/components/landing/EventsViewMode/EventsViewMode.js]: Do you need it?

File: src/components/landing/landing.js [hunk 8]
```diff
@@ -1,52 +1,47 @@
...
 
-  handleClick = () => {
-    this.props.onSubmit();
-  };
+const Landing = ({ getUpcomingEventsDispatch, events }) => {
+  const [eventsViewMode, setEventsViewMode] = useState("slider");
+
+  useEffect(() => {
+    getUpcomingEventsDispatch();
+  }, [getUpcomingEventsDispatch]);
```
Comment (andrii-kuzmichov) [inline src/components/landing/landing.js]: It is not the same as `componentDidMount`

File: src/components/landing/landing.js [hunk 9]
```diff
@@ -1,52 +1,47 @@
...
+
+  // nowhere used
```
Comment (andrii-kuzmichov) [inline src/components/landing/landing.js:22]: Add `TODO` title for easy finding

Comment (Stanislav-Vyhnan) [reply src/components/landing/landing.js:22]: Added `TODO` ✔

File: src/components/landing/landing.js (hunk 8 above)
Comment (Stanislav-Vyhnan) [reply src/components/landing/landing.js]: Removed dependencies, left an empty array ✔

File: src/components/landing/EventsViewMode/EventsViewMode.js (hunk 7 above)
Comment (Stanislav-Vyhnan) [reply src/components/landing/EventsViewMode/EventsViewMode.js]: Yeap, it fixes eslint error `A form label must be associated with a control`

File: src/components/landing/EventsViewMode/EventsViewMode.js (hunk 6 above)
Comment (Stanislav-Vyhnan) [reply src/components/landing/EventsViewMode/EventsViewMode.js]: Moved ✔

File: src/components/landing/EventsListItem/EventListCard/EventListCard.js (hunk 5 above)
Comment (Stanislav-Vyhnan) [reply src/components/landing/EventsListItem/EventListCard/EventListCard.js]: I completely agree. Changed ✔

File: src/components/landing/EventsListItem/EventListCard/EventListCard.js (hunk 4 above)
Comment (Stanislav-Vyhnan) [reply src/components/landing/EventsListItem/EventListCard/EventListCard.js]: Combined into one div ✔

File: src/components/landing/EventsListItem/EventListCard/EventListCard.js (hunk 3 above)
Comment (Stanislav-Vyhnan) [reply src/components/landing/EventsListItem/EventListCard/EventListCard.js]: I don't know. It was written here before us, but I added this check ✔

File: src/components/helpers/landingUtils.js (hunk 2 above)
Comment (Stanislav-Vyhnan) [reply src/components/helpers/landingUtils.js]: Refactored ✔

File: src/components/helpers/landingUtils.js (hunk 1 above)
Comment (Stanislav-Vyhnan) [reply src/components/helpers/landingUtils.js]: I created an object for it. In my opinion it's no longer needed

**Review by andrii-kuzmichov — APPROVED** [review]

**Pull request merged**